Missionframework/modules/06_virtual/fnc/fn_virtual_addCurator.sqf

```diff
@@ -0,0 +1,84 @@
+/*
+    KPLIB_fnc_virtual_addCurator
+
+    File: fn_virtual_addCurator.sqf
+    Author: KP Liberation Dev Team - https://github.com/KillahPotatoes
+    Date: 2018-11-18
+    Last Update: 2018-11-27
+    License: GNU General Public License v3.0 - https://www.gnu.org/licenses/gpl-3.0.html
+
+    Description:
+        Adds curator cababilities to unit.
+
+    Parameter(s):
+        _unit - Unit to add curator for                                             [OBJECT, defaults to objNull]
+        _mode - Curator mode 1 (limited) or 2 (limited, free camera) or 3 (full)    [NUMBER, defaults to 1 (limited)]
+
+    Returns:
+        Curator logic [OBJECT]
+*/
+params [
+    ["_unit", objNull, [objNull]],
+    ["_mode", 1, [0]]
+];
+
+// Do not add curator if mode out of range or unit is null
+if (!(_mode in [1, 2, 3]) || isNull _unit) exitWith {objNull};
+
+if (KPLIB_param_debug) then {diag_log format ["[KP LIBERATION] [VIRTUAL] Adding curator for unit '%1' with mode %2", _unit, _mode]};
+
+_unit call KPLIB_fnc_virtual_removeCurator;
+
+private _curator = (createGroup sideLogic) createUnit ["ModuleCurator_F", [0, 0, 0], [], 0, "CAN_COLLIDE"];
+// Add player so he can see himself in curator
+_curator addCuratorEditableObjects [[_unit], false];
+
+_curator setVariable ["KPLIB_mode", _mode];
+
+switch _mode do {
+    // Limited mode, limited and full camera
+    case 1;
+    case 2: {
+        _curator setVariable ["Addons", 0, true];
+
+        // Add Built items to zeus
+        private _fobSaveNamespace = (missionNamespace getVariable ["KPLIB_build_saveNamespace", objNull]);
+        private _fobsItems = [];
+        {
+            _fobsItems append (_fobSaveNamespace getVariable _x);
+        } forEach allVariables _fobSaveNamespace;
+
+        _curator addCuratorEditableObjects [_fobsItems, true];
+
+        // Disable place, edit, delete and destroy
+        {
+            _curator setCuratorCoef [_x, -1e10];
+        } forEach ["place", "edit", "delete", "destroy"];
+
+        // Disallow attributes changing of objects
+        [
+            _curator,
+            "object",
+            []
+        ] call BIS_fnc_setCuratorAttributes;
+    };
+    // Full mode
+    case 3: {
+        _curator setVariable ["Addons", 3, true];
+        _curator addCuratorEditableObjects [entities "", true];
```

Comment:
This would also include every enemy/civilian/resistance unit.
It's "full zeus", of course. But I guess people would like to be able to do everything but not get spoiled. Maybe a 4th mode?
3 -> Full functionalities, but auto add is limited to friendly units
4 -> Full functionalities, everything gets added

Reply:
I would consider adding it later in next Sprint, I've already done 200% of what I initially had to do with this module. Are you ok with that?

Reply:
Oh, sorry. It wasn't meant like "add now a 4th mode", it was meant in general 👍